### PR TITLE
Add linting

### LINKED ISSUE #6: Add linting
ruff+mypy has worked in other projects like python-tuf, that might be good here as well. Those projects use tox (https://github.com/theupdateframework/python-tuf/blob/main/tox.ini) but a Makefile should work as well

### PR DESCRIPTION
**Pull request by jku**
* Add ruff and mypy as optional dependencies
* Modify Makefile so it can install linters and has targets `lint` and `fix`
* Modify CI workflow so it runs `make lint`
* Mention linting in README
* Fix all issues that linters complain about
* Remove some dead code

This is only a start: the ruff default rulesets are not very strict but this initial commit already improves code quality so clearly this is useful.

Fixes #6 , should be ready for review now

### CODE REVIEW

File: tuf_conformance/metadata.py [hunk 1]
```diff
@@ -28,7 +29,7 @@
 
 class MetadataTest(Metadata):
     @classmethod
-    def from_dict(cls, metadata: Dict[str, Any]) -> "MetadataTest[T]":
+    def from_dict(cls, metadata: Dict[str, Any]) -> "Metadata[T]":
```
Comment (AdamKorcz) [inline tuf_conformance/metadata.py:32]: What is the reason for this change?

File: tuf_conformance/metadata.py [hunk 2]
```diff
@@ -109,7 +110,7 @@ def __init__(
 class JSONDeserializerTest(MetadataDeserializer):
     """Provides JSON to Metadata deserialize method."""
 
-    def deserialize(self, raw_data: bytes) -> MetadataTest:
+    def deserialize(self, raw_data: bytes) -> Metadata:
```
Comment (AdamKorcz) [inline tuf_conformance/metadata.py:113]: Same as https://github.com/theupdateframework/tuf-conformance/pull/109/files#r1704294902

File: tuf_conformance/metadata.py (hunk 1 above)
Comment (jku) [reply tuf_conformance/metadata.py:32]: mypy complains about this type annotation when the method is called.. and I believe it is correct to complain: we do return a Metadata from the is method because we end up calling `Metadata()` at the end of the function.

Comment (jku) [reply tuf_conformance/metadata.py:32]: I could be wrong about the logic there, python inheritance has never been my strongest suit

Comment (jku) [reply tuf_conformance/metadata.py:32]: the error is 
> tuf_conformance/metadata.py:32: error: "MetadataTest" expects no type arguments, but 1 given  [type-arg]

which admittedly is a bit strange.

Comment (jku) [reply tuf_conformance/metadata.py:32]: hmmm... I think this might be related to the generic type, let me try a different tweak to fix this

Comment (jku) [reply tuf_conformance/metadata.py]: Yeah tweaking the inheritance annotation `class MetadataTest(Metadata[T])` makes mypy understand the intent as well. Fixed with a new commit.

**Review by AdamKorcz — APPROVED** [review]
